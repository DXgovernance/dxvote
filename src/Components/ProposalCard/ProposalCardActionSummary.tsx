import { Loading } from 'Components/Primitives/Loading';
import { getInfoLineView } from 'old-components/Guilds/ActionsBuilder/SupportedActions';
import UndecodableCallInfoLine from 'old-components/Guilds/ActionsBuilder/UndecodableCalls/UndecodableCallsInfoLine';
import {
  ActionsWrapper,
  BorderedIconDetailWrapper,
  NotFoundActionWrapper,
} from 'Components/ProposalCard/ProposalCard.styled';

interface ProposalCardActionSummaryProps {
  actions?: any[];
}

const ProposalCardActionSummary: React.FC<ProposalCardActionSummaryProps> = ({
  actions,
}) => {
  const isLoading = !actions;
  if (isLoading) {
    return (
      <Loading
        style={{ margin: 0 }}
        loading
        text
        skeletonProps={{ width: '200px' }}
      />
    );
  }
  return (
    <ActionsWrapper>
      {actions?.map(action => {
        if (!action) return null;
        const InfoLine = getInfoLineView(action?.decodedCall?.callType);

        return !!InfoLine ? (
          <BorderedIconDetailWrapper>
            <InfoLine
              decodedCall={action?.decodedCall}
              approveSpendTokens={action?.approval}
              compact
            />
          </BorderedIconDetailWrapper>
        ) : (
          <NotFoundActionWrapper>
            <UndecodableCallInfoLine />
          </NotFoundActionWrapper>
        );
      })}
    </ActionsWrapper>
  );
};

export default ProposalCardActionSummary;
