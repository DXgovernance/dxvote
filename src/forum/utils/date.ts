import moment from 'moment';

export const formatTime = date => {
  const d = new Date(date);
  return moment(d).format('MMM dd');
};

