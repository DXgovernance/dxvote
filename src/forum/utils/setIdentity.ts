import Identities from 'orbit-db-identity-provider';
import { getResolver } from '@ceramicnetwork/3id-did-resolver';

// Connects the logged in Ceramic user to OrbitDB
export async function setIdentity({ orbitdb, registry }, { ceramic, threeId }) {
  if (orbitdb.identity.type !== 'did') {
    Identities.DIDIdentityProvider.setDIDResolver(getResolver(ceramic));
    const identity = await Identities.createIdentity({
      type: 'DID',
      didProvider: threeId.getDidProvider(),
    });
    orbitdb.identity = identity;
    registry.setIdentity(identity);
    console.log('OrbitDB: DID identity set!', identity);
  }
}
