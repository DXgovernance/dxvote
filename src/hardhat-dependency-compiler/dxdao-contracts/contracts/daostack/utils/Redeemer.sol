// SPDX-License-Identifier: UNLICENSED
pragma solidity >0.0.0;
import 'dxdao-contracts/contracts/daostack/utils/Redeemer.sol';
