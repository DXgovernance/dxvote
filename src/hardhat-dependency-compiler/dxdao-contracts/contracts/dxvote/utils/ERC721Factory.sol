// SPDX-License-Identifier: UNLICENSED
pragma solidity >0.0.0;
import 'dxdao-contracts/contracts/dxvote/utils/ERC721Factory.sol';
