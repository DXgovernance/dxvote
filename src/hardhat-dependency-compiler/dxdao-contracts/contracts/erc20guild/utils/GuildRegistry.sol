// SPDX-License-Identifier: UNLICENSED
pragma solidity >0.0.0;
import 'dxdao-contracts/contracts/erc20guild/utils/GuildRegistry.sol';
