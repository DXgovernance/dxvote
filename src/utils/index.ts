export * from './address';
export * from './bignumber';
export * from './cache';
export * from './constants';
export * from './date';
export * from './enums';
export * from './helperHooks';
export * from './helpers';
export * from './number';
export * from './permissions';
export * from './proposals';
export * from './scheme';
export * from './transaction';
export * from './encodingCalls';
export * from './signature';
export * from './array';
export * from './votes';
