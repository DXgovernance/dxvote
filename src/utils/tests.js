import { ThemeProvider } from 'styled-components';
import { render as rtlRender } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { GuildsDarkTheme } from 'Components/theme';

export const renderWithRoutes = (ui, options = {}) => {
  //wrap with MemoryRouter
  return rtlRender(ui, { wrapper: MemoryRouter, ...options });
};

// we are mokcing testing-library/render with our own render
export const render = (children, options = {}) => {
  //wrap with Theme
  return renderWithRoutes(
    <ThemeProvider theme={GuildsDarkTheme}> {children} </ThemeProvider>,
    options
  );
};
