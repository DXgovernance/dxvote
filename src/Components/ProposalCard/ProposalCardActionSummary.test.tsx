import ProposalCardActionSummary from './ProposalCardActionSummary';
import { render } from '../../utils/tests';

jest.mock('ipfs', () => jest.fn());
jest.mock('cids', () => jest.fn());
jest.mock('axios', () => jest.fn());
jest.mock('hooks/Guilds/guild/useFilteredProposalActions', () => {
  return {
    useFilteredProposalActions: jest.fn(),
  };
});
jest.mock('contexts/index', () => jest.fn());

describe('ProposalCardActionSummary', () => {
  it('Should match snapshot without proposal', async () => {
    const { container } = render(<ProposalCardActionSummary proposal={null} />);
    expect(container).toMatchSnapshot();
  });
});
