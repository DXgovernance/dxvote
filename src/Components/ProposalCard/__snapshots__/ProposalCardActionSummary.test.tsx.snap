// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalCardActionSummary Should match snapshot without proposal 1`] = `
<div>
   
  <div
    class="sc-jrAFXE sc-ehSDrC jLAWlL cclvjV"
  />
   
</div>
`;
